src/pages/journal/journal.ts

```diff
@@ -104,22 +119,42 @@ export class JournalPage extends BasePageComponent implements OnInit, OnDestroy
     this.subscription.unsubscribe();
   }
 
+  ionViewWillEnter() {
+    super.ionViewWillEnter();
+    this.loadJournalManually();
+    this.setupPolling();
+    return true;
+  }
+
+  ionViewWillLeave() {
+    this.store$.dispatch(new journalActions.StopPolling());
+  }
+
   ionViewDidEnter(): void {
     this.analytics.setCurrentPage(AnalyticsScreenNames.JOURNAL);
   }
 
-  loadJournal() {
+  loadJournalManually() {
     this.store$.dispatch(new journalActions.LoadJournal());
-    this.createLoadingSpinner();
+  }
+
+  setupPolling() {
+    this.store$.dispatch(new journalActions.SetupPolling());
   }
 
   handleLoadingUI = (isLoading: boolean): void => {
-    if (!isLoading) {
-      this.pageRefresher ? this.pageRefresher.complete() : null;
-      if (this.loadingSpinner) {
-        this.loadingSpinner.dismiss();
-        this.loadingSpinner = null;
-      }
+    if (isLoading) {
+      this.loadingSpinner = this.loadingController.create({
+        dismissOnPageChange: true,
+        spinner: 'circles'
+      });
+      this.loadingSpinner.present();
+      return;
+    }
+    this.pageRefresher ? this.pageRefresher.complete() : null;
```

Comment:
`this.pageRefresher || this.pageRefresher.complete();` would do the same operation.